optimum/bettertransformer/models/encoder_models.py

```diff
@@ -751,3 +751,111 @@ def forward(self, hidden_states, attention_mask, **__):
         if hidden_states.is_nested and self.is_last_layer:
             hidden_states = hidden_states.to_padded_tensor(0.0)
         return (hidden_states,)
+
+
+class FSMTEncoderLayerBetterTransformer(BetterTransformerBaseLayer):
+    def __init__(self, fsmt_layer, config):
+        r"""
+                A simple conversion of the FSMT Encoder layer to its `BetterTransformer` implementation.
+
+                Args:
+                    fsmt_layer (`torch.nn.Module`):
+                        The original FSMT Layer where the weights needs to be retrieved.
+                """
+        super().__init__(config)
+        # In_proj layer
+        self.in_proj_weight = nn.Parameter(
+            torch.cat(
+                [
+                    fsmt_layer.self_attn.q_proj.weight,
+                    fsmt_layer.self_attn.k_proj.weight,
+                    fsmt_layer.self_attn.v_proj.weight,
+                ]
+            )
+        )
+        self.in_proj_bias = nn.Parameter(
+            torch.cat(
+                [
+                    fsmt_layer.self_attn.q_proj.bias,
+                    fsmt_layer.self_attn.k_proj.bias,
+                    fsmt_layer.self_attn.v_proj.bias,
+                ]
+            )
+        )
+
+        # Out proj layer
+        self.out_proj_weight = fsmt_layer.self_attn.out_proj.dense.weight
+        self.out_proj_bias = fsmt_layer.self_attn.out_proj.dense.bias
+
+        # Linear layer 1
+        self.linear1_weight = fsmt_layer.fc1.weight
+        self.linear1_bias = fsmt_layer.fc1.bias
+
+        # Linear layer 2
+        self.linear2_weight = fsmt_layer.fc2.weight
+        self.linear2_bias = fsmt_layer.fc2.bias
+
+        # Layer norm 1
+        self.norm1_eps = fsmt_layer.self_attn_layer_norm.eps
+        self.norm1_weight = fsmt_layer.self_attn_layer_norm.weight
+        self.norm1_bias = fsmt_layer.self_attn_layer_norm.bias
+
+        # Layer norm 2
+        self.norm2_eps = fsmt_layer.final_layer_norm.eps
+        self.norm2_weight = fsmt_layer.final_layer_norm.weight
+        self.norm2_bias = fsmt_layer.final_layer_norm.bias
+
+        # Model hyper parameters
+        self.num_heads = fsmt_layer.self_attn.num_heads
+        self.embed_dim = fsmt_layer.self_attn.embed_dim
+
+        # Last step: set the last layer to `False` -> this will be set to `True` when converting the model
+        self.is_last_layer = False
```

Comment:
It seems that the conversion is still not succesfull, you might need to review that once again. Could you make sure your conversion runs fine with the following script:

```
from transformers import AutoModel
from optimum.bettertransformer import BetterTransformer

model_id = "hf-internal-testing/tiny-random-FSMTModel"

hf_model = AutoModel.from_pretrained(model_id)
bt_model = BetterTransformer.transform(hf_model)
```

From there you will have different errors that you will need to dig a little bit and find out which key is not correct.